### PR TITLE
feat: Suppress zero fills on deposits with messages

### PR DESCRIPTION
**Pull request by pxrl**

### CODE REVIEW

**Comment by pxrl** [comment]
todo: follow up with a small test to verify no zero-fills.

**Review by nicholaspai — COMMENTED** [review]
Should we also check that the recipient is a contract or is that overkill? Without this check we might get some false negatives but it shouldn't happen often in prod

**Review by nicholaspai — APPROVED** [review]

**Comment by pxrl** [comment]
> Should we also check that the recipient is a contract or is that overkill? Without this check we might get some false negatives but it shouldn't happen often in prod

Hm, so you're thinking about the scenario where the deposit has a message attached, and the recipient is an EOA, so we'd actually be OK with initiating a slow fill for that?

File: src/relayer/Relayer.ts [hunk 1]
```diff
@@ -352,6 +352,14 @@ export class Relayer {
    * @param deposit Deposit object to zero-fill.
    */
   zeroFillDeposit(deposit: Deposit): void {
+    if (!isMessageEmpty(deposit.message)) {
```
Comment (nicholaspai) [inline src/relayer/Relayer.ts:355]: Can you add a comment that if the depositor has sped up their message and zero'd their message, we still won't partial fill their deposit, but we can now full fill it

Comment (james-a-morris) [inline src/relayer/Relayer.ts:355]: Good call

**Review by james-a-morris — APPROVED** [review]

File: src/relayer/Relayer.ts (hunk 1 above)
Comment (pxrl) [reply src/relayer/Relayer.ts:357]: Updated: https://github.com/across-protocol/relayer-v2/pull/1042/commits/99e2b65d80c7e17365eb3f49bff3df62ee7eed7b

**Pull request merged**